🐛 BUG: --from-dash confusing

### What version of `Wrangler` are you using?

latest

### What operating system are you using?

All

### Describe the Bug

I ran `--from-dash` to see what it does, and found the output confusing:

```
rozenmd@QRVX0PWXHR Testing % npx wrangler@beta init test --from-dash

✘ [ERROR] Not enough arguments following: from-dash


If you think this is a bug then please create an issue at https://github.com/cloudflare/wrangler2/issues/new/choose
```

Running `--help` here then gives you:
```
rozenmd@QRVX0PWXHR Testing % npx wrangler@beta init test --from-dash --help
wrangler init [name]

📥 Create a wrangler.toml configuration file

Positionals:
  name  The name of your worker  [string]

Flags:
  -c, --config   Path to .toml configuration file  [string]
  -e, --env      Environment to use for operations and .env files  [string]
  -h, --help     Show help  [boolean]
  -v, --version  Show version number  [boolean]

Options:
  -y, --yes        Answer "yes" to any prompts for new projects  [boolean]
      --from-dash  Download script from the dashboard for local development  [string]
```

Which doesn't tell me what from-dash wants from me.

--- a/.changeset/tough-carrots-live.md
+++ b/.changeset/tough-carrots-live.md
@@ -1,0 +1,10 @@
+---
+"wrangler": patch
+---
+
+fix: make --from-dash error output clearer
+
+This PR makes it clearer what --from-dash wants from you.
+
+closes #2373
+closes #2375

--- a/packages/wrangler/src/__tests__/init.test.ts
+++ b/packages/wrangler/src/__tests__/init.test.ts
@@ -2831,7 +2831,7 @@ describe("init", () => {
 			await expect(
 				runWrangler("init  --from-dash")
 			).rejects.toMatchInlineSnapshot(
-				`[YError: Not enough arguments following: from-dash]`
+				`[Error: Not enough arguments following: from-dash]`
 			);
 			checkFiles({
 				items: {

--- a/packages/wrangler/src/deprecated/index.ts
+++ b/packages/wrangler/src/deprecated/index.ts
@@ -14,21 +14,16 @@ import type {
 	CommonYargsOptions,
 	YargsOptionsToInterface,
 } from "../yargs-types";
-import type { Argv, ArgumentsCamelCase, BuilderCallback } from "yargs";
+import type { Argv, BuilderCallback } from "yargs";
 
 // https://github.com/cloudflare/wrangler/blob/master/src/cli/mod.rs#L106-L123
-interface GenerateArgs {
-	name?: string;
-	template?: string;
-	type?: string;
-	site?: boolean;
-}
 
 export function generateOptions(yargs: Argv<CommonYargsOptions>) {
 	return yargs
 		.positional("name", {
 			describe: "Name of the Workers project",
 			type: "string",
+			demandOption: true,
 		})
 		.positional("template", {
 			type: "string",
@@ -47,25 +42,34 @@ export function generateOptions(yargs: Argv<CommonYargsOptions>) {
 			deprecated: true,
 		});
 }
+type GenerateArgs = YargsOptionsToInterface<typeof generateOptions>;
 
-export async function generateHandler({
+//TODO: move this out of deprecated since apparently it's not deprecated anymore
+export async function generateHandler(args: GenerateArgs) {
 	// somehow, `init` marks name as required but then also runs fine
 	// with the name omitted, and then substitutes it at runtime with ""
-	name = "",
-	template,
-	type,
-	site,
-	...args
-}: ArgumentsCamelCase<GenerateArgs>) {
 	// delegate to `wrangler init` if no template is specified
-	if (template === undefined) {
-		return initHandler({ name, ...args });
+	if (args.template === undefined) {
+		return initHandler({
+			name: args.name,
+			template: undefined,
+			site: undefined,
+			yes: undefined,
+			fromDash: undefined,
+			"from-dash": undefined,
+			v: undefined,
+			config: undefined,
+			env: undefined,
+			type: undefined,
+			_: args._,
+			$0: args.$0,
+		});
 	}
 
 	// print down here cuz `init` prints it own its own
 	await printWranglerBanner();
 
-	if (type) {
+	if (args.type) {
 		let message = "The --type option is no longer supported.";
 		if (args.type === "webpack") {
 			message +=
@@ -75,13 +79,13 @@ export async function generateHandler({
 		throw new CommandLineArgsError(message);
 	}
 
-	if (isRemote(name)) {
-		[template, name] = [name, template];
+	if (args.name && isRemote(args.name)) {
+		[args.template, args.name] = [args.name, args.template];
 	}
 
-	const creationDirectory = generateWorkerDirectoryName(name);
+	const creationDirectory = generateWorkerDirectoryName(args.name);
 
-	if (site) {
+	if (args.site) {
 		const gitDirectory =
 			creationDirectory !== process.cwd()
 				? path.basename(creationDirectory)
@@ -99,10 +103,12 @@ export async function generateHandler({
 	}
 
 	logger.log(
-		`Creating a worker in ${path.basename(creationDirectory)} from ${template}`
+		`Creating a worker in ${path.basename(creationDirectory)} from ${
+			args.template
+		}`
 	);
 
-	const { remote, subdirectory } = parseTemplatePath(template);
+	const { remote, subdirectory } = parseTemplatePath(args.template);
 
 	await cloneIntoDirectory(remote, creationDirectory, subdirectory);
 	await initializeGit(creationDirectory);

--- a/packages/wrangler/src/init.ts
+++ b/packages/wrangler/src/init.ts
@@ -22,10 +22,13 @@ import type { Route, SimpleRoute } from "./config/environment";
 import type { WorkerMetadata } from "./create-worker-upload-form";
 import type { ConfigPath } from "./index";
 import type { PackageManager } from "./package-manager";
-import type { CommonYargsOptions } from "./yargs-types";
-import type { Argv, ArgumentsCamelCase } from "yargs";
+import type {
+	CommonYargsOptions,
+	YargsOptionsToInterface,
+} from "./yargs-types";
+import type { Argv } from "yargs";
 
-export async function initOptions(yargs: Argv<CommonYargsOptions>) {
+export function initOptions(yargs: Argv<CommonYargsOptions>) {
 	return yargs
 		.positional("name", {
 			describe: "The name of your worker",
@@ -56,12 +59,7 @@ export async function initOptions(yargs: Argv<CommonYargsOptions>) {
 		});
 }
 
-interface InitArgs {
-	name: string;
-	type?: string;
-	site?: boolean;
-	yes?: boolean;
-}
+type InitArgs = YargsOptionsToInterface<typeof initOptions>;
 
 export type ServiceMetadataRes = {
 	id: string;
@@ -110,7 +108,7 @@ export type CronTriggersRes = {
 	];
 };
 
-export async function initHandler(args: ArgumentsCamelCase<InitArgs>) {
+export async function initHandler(args: InitArgs) {
 	await printWranglerBanner();
 	if (args.type) {
 		let message = "The --type option is no longer supported.";
@@ -125,7 +123,7 @@ export async function initHandler(args: ArgumentsCamelCase<InitArgs>) {
 	const devDepsToInstall: string[] = [];
 	const instructions: string[] = [];
 	let shouldRunPackageManagerInstall = false;
-	const fromDashScriptName = args["from-dash"] as string;
+	const fromDashScriptName = args.fromDash;
 	const creationDirectory = path.resolve(
 		process.cwd(),
 		(args.name ? args.name : fromDashScriptName) ?? ""

